src/custom/hooks/useChangeNetworks.ts

```diff
@@ -1,82 +1,113 @@
-import { useCallback, useEffect, useRef } from 'react'
+import { useCallback, useEffect, useMemo, useRef, useState } from 'react'
 import { useOnClickOutside } from 'hooks/useOnClickOutside'
 import { useActiveWeb3React } from 'hooks/web3'
 import { switchToNetwork } from 'utils/switchToNetwork'
-import { useModalOpen, useToggleModal } from '../state/application/hooks'
+import { useModalOpen, useToggleModal, useWalletModalToggle } from 'state/application/hooks'
 import { CHAIN_INFO } from 'constants/chainInfo'
-import useParsedQueryString from '@src/hooks/useParsedQueryString'
-import usePrevious from '@src/hooks/usePrevious'
-import { addPopup, ApplicationModal } from '@src/state/application/reducer'
-import { useAppDispatch } from '@src/state/hooks'
-import { replaceURLParam } from '@src/utils/routes'
+import useParsedQueryString from 'hooks/useParsedQueryString'
+import usePrevious from 'hooks/usePrevious'
+import { addPopup, ApplicationModal } from 'state/application/reducer'
+import { useAppDispatch } from 'state/hooks'
+import { replaceURLParam } from 'utils/routes'
 import { getChainNameFromId, getParsedChainId } from 'components/Header/NetworkSelector'
 import { useHistory } from 'react-router-dom'
+import { supportedChainId } from 'utils/supportedChainId'
 
-type ChangeNetworksParams = Pick<ReturnType<typeof useActiveWeb3React>, 'chainId' | 'library'>
+type ChangeNetworksParams = Pick<ReturnType<typeof useActiveWeb3React>, 'account' | 'chainId' | 'library'>
+export type ChainSwitchCallbackOptions = { skipWalletToggle: boolean; skipToggle: boolean }
 
 /**
  * Hook extracted from Header/NetworkSelector component pretty much verbatim
  *
  * @param chainId
  * @param library
  */
-export default function useChangeNetworks({ chainId, library }: ChangeNetworksParams) {
+export default function useChangeNetworks({ account, chainId: preChainId, library }: ChangeNetworksParams) {
   const parsedQs = useParsedQueryString()
   const { urlChain, urlChainId } = getParsedChainId(parsedQs)
-  const prevChainId = usePrevious(chainId)
+  const prevChainId = usePrevious(preChainId)
   const node = useRef<HTMLDivElement>()
   const open = useModalOpen(ApplicationModal.NETWORK_SELECTOR)
   const toggle = useToggleModal(ApplicationModal.NETWORK_SELECTOR)
   useOnClickOutside(node, open ? toggle : undefined)
 
   const history = useHistory()
 
-  const info = chainId ? CHAIN_INFO[chainId] : undefined
-
   const dispatch = useAppDispatch()
 
+  const toggleWalletModal = useWalletModalToggle() // MOD
+  const [queuedNetworkSwitch, setQueuedNetworkSwitch] = useState<null | number>(null) // MOD
+  // MOD: get supported chain and check unsupported
+  const chainId = useMemo(() => {
+    const chainId = supportedChainId(preChainId)
+
+    return chainId
+  }, [preChainId])
```

Comment:
Nitpick: refactoring

```suggestion
  const chainId = useMemo(() => supportedChainId(preChainId), [preChainId])
```